### DISCUSSION TITLE
Is there a way for a select field to refresh it's options based on the value in another field?

### DISCUSSION THREAD

**Discussion by madalinignisca** [area17/twill#2595] [Q&A]
Is there a way for a select field to refresh it's options based on the value in another field?

My case is querying an external API, I am getting first category IDs. With an ID, I must select from it's list of events.

I know how to get the values right now, but I could not figure out any event of twill's fields that would re-render a field based on change of another field.

**Comment by ifox** [comment]
This is possible with connected select fields if you can pre-compute all values to conditionally add other select fields based on the first, but obviously with your use case it doesn't really scale.

It is also possible with connected browser fields though, as the connected one can receive the selected value of the first one. Since the browser records are fetched asynchronously, you can request the external API there.

Lastly, you could build a custom form field with a vue component if browsers are not working for you. 

Supporting dynamic connected select fields would be a good addition to our roadmap.

**Reply by madalinignisca** [reply]
Adding this would be great. For web applications, with fillamentphp with been using this feature from early stage. Their approach is quite easy to understand for end-developers building apps.
